### PR TITLE
fix: set credential as password

### PR DESCRIPTION
**Pull request by SMillerDev**
A very naive fix for https://github.com/1Password/terraform-provider-onepassword/issues/52

### CODE REVIEW

**Comment by asininemonkey** [comment]
Could either @volodymyrZotov or @jpcoenen please review/merge this? This is a major hindrance whilst item type parity is lacking.

**Comment by volodymyrZotov** [comment]
@SMillerDev I looked at it; unfortunately, this implementation doesn't work. The `field.Purpose` value is always empty for `api_credential` fields. 

You can use `field.ID` to check that there is a `credential` field:
```
if field.ID == "credential" {
     data.Set("password", field.Value)
} 
```

However, that would fix the problem only partially.
The thing is that the `ID` value for `credential` field is mutable, `purpose` is always empty and there is no other field that we can rely on to make sure the right field is found. If you ever change the `ID` (for example using 1Passsword Connect), the provider won't be able to find it. We need to make some improvements on our side to fix that.

To summarise, let's check `field.ID == "credential"` for now to make it work and I'll raise that within the team so we decide what we'll do on our end.

File: onepassword/data_source_onepassword_item.go [hunk 1]
```diff
@@ -214,6 +214,10 @@ func dataSourceOnepasswordItemRead(ctx context.Context, data *schema.ResourceDat
 			data.Set("username", f.Value)
 		case "PASSWORD":
 			data.Set("password", f.Value)
+		case "CREDENTIAL":
+			if strings.ToLower(string(item.Category)) == "api_credential" {
```
Comment (SMillerDev) [inline onepassword/data_source_onepassword_item.go:218]:
```suggestion
			if f.ID == "credential" {
```

**Comment by SMillerDev** [comment]
Fixed, let me know what else

**Comment by volodymyrZotov** [comment]
@SMillerDev The change is inside `switch` statement, which checks `f.Purpose`. The `f.Purpose` for the credentials field will always be an empty string. So, it'll never reach `if f.ID == "credential" {` statement.

Let's put `if f.ID == "credential" {` after the `switch`.

**Comment by volodymyrZotov** [comment]
@SMillerDev Thinking more about this, there are several questions I have.

1. Why set the credential into `password` schema field? Why not create a new `credentials` property in the schema? aka
```
"credential": {
    Description:  "Some description",
    Type:         schema.TypeString,
    Optional:     true,
    Computed:     true,
    Sensitive:    true,
},
```
So then in to code toy would set it like `data.Set("credential", f.Value)`

2. I'd extend the `if` statement a bit more, to make sure we are looking for `credential` field only if item category is `api_credential`, aka
```
if f.ID == "credential" && item.Category == "API_CREDENTIAL" {
    data.Set("credential", f.Value)
}
```

**Comment by SMillerDev** [comment]
Those all sound fine. I didn't really want to get too invasive with my changes so I kept them small, but happy to tweak them like that

**Comment by volodymyrZotov** [comment]
I'd appreciate it if you could do that.

I have another question though. The changes are only for data-source. Is it possible to do the same for the resource types?
I'd like to include that also to keep the consistency between data-source and resources.

**Comment by SMillerDev** [comment]
If this one works for you I'll add it to the resource too.

**Comment by volodymyrZotov** [comment]
@SMillerDev, sorry for the delay. A migration happened in the background https://github.com/1Password/terraform-provider-onepassword/pull/167 that led to conflicts with your work. Do have a chance to resolve them, please?

**Comment by SMillerDev** [comment]
No problem, it's really not that much time I've spend on this. I'll check again later

**Comment by SMillerDev** [comment]
Okay, updated with the latest changes. Let me know if this is good and I'll add it to the resource.

**Comment by volodymyrZotov** [comment]
Hi @SMillerDev . Had a chance to test it. It works fine.👍
I see that ci job for documentation doesn't pass. You can fix it by running the following command.
```
go run github.com/hashicorp/terraform-plugin-docs/cmd/tfplugindocs generate -provider-name onepassword
```
Then just push updated documentation.

**Comment by SMillerDev** [comment]
I think it would certainly make it faster if you can give some instructions, I don't have that much time (or interest) in a deep dive into tf provider testing. 

You're also very welcome to push to my branch or just copy my changes if that's easier. I'm not doing this for my portfolio/credits, I'd just like to use credentials in the data source 😁

**Review by volodymyrZotov — APPROVED** [review]
✅ Approved it.

You can fetch 'credential' items using data-source now. The resource is not supported in this PR and can be added in a separate one.
